### PR TITLE
Add new `Lint/SafeNavigationChain` cop

### PR DESCRIPTION
**Pull request by pocke**
Safe navigation operator returns nil if the receiver is nil.
So if you chain an ordinary method call after safe navigation operator, it raises NoMethodError.
We should use safe navigation operator after safe navigation operator.
This cop checks the problem.

For example

```ruby
x = nil
x&.foo.bar   # => NoMethodError: undefined method `bar' for nil:NilClass
x&.foo&.bar  # => nil
```

-----------------

Before submitting the PR make sure the following are checked:

* [x] Wrote [good commit messages][1].
* [ ] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Used the same coding conventions as the rest of the project.
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Added an entry to the [Changelog](../blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](../blob/master/CONTRIBUTING.md#changelog-entry-format).
* [x] All tests are passing.
* [x] The new code doesn't generate RuboCop offenses.
* [x] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences.
* [x] Updated cop documentation with `rake generate_cops_documentation` (required only when you've added a new cop or changed the configuration/documentation of an existing cop).

[1]: http://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html

### CODE REVIEW

**Comment by bbatsov** [comment]
Another great contribution! Thanks! 👍 ❤️ 🚀

File: lib/rubocop/cop/lint/safe_navigation_chain.rb [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+            return if nil_methods.include?(method)
+
+            loc = node.loc.dot || :selector
+            add_offense(node, loc)
+          end
+        end
+
+        def autocorrect(node)
+          dot = node.loc.dot
+          return unless node.loc.dot
```
Comment (Drenmi) [inline lib/rubocop/cop/lint/safe_navigation_chain.rb]: It's a bit late now, but I guess we could have used the local variable `dot` here. 😀

Comment (bbatsov) [inline lib/rubocop/cop/lint/safe_navigation_chain.rb]: I'll fix this myself. :-)

Comment (pocke) [inline lib/rubocop/cop/lint/safe_navigation_chain.rb]: 🙀 🙀 Thanks! 🙇

**Pull request merged**